### PR TITLE
Categorize types

### LINKED ISSUE #25: Categorize types
We're starting to get quite a few types, and they're all defined in one types.ts file. And it will get even worse if we decide to use Redux for storing API data and need types for actions and action types and payloads.

### PR DESCRIPTION
**Pull request by erikskaar**
closes #25

### CODE REVIEW

**Comment by erikskaar** [comment]
Not sure about the naming scheme at `apiTypes.ts`, so feel free to come with suggestions that might make more sense :)

File: frontend/src/types/apiTypes.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+}
+
+export type ErrorState = {
+  apiError: ApiError | null;
+};
+
+export type SetErrorStateAction = {
+  type: typeof SET_ERROR;
+  payload: ApiError;
```
Comment (AndreasWintherMoen) [inline frontend/src/types/apiTypes.ts:19]: If this is meant to be the way to display all errors we should probably have just Error be the payload type. Can just rename the file to errorTypes.ts as well.

File: frontend/src/types/apiTypes.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  constructor(res: any, body: any) {
+    super(body.message || res.statusText);
+    this.status = res.status;
+    this.name = 'ApiError';
+    this.body = body;
+  }
+}
+
+export type ErrorState = {
+  apiError: ApiError | null;
```
Comment (AndreasWintherMoen) [inline frontend/src/types/apiTypes.ts:14]: Same here. Should just be error, not apiError.

**Review by AndreasWintherMoen — COMMENTED** [review]
I suggest we do this error handling for all types of errors, not just api errors. See comments. Other than that it looks good.

**Review by AndreasWintherMoen — APPROVED** [review]
Looking good

**Pull request merged**